src/Microsoft.PowerApps.TestEngine/Helpers/PollingHelper.cs

```diff
@@ -29,6 +29,42 @@ public static T Poll<T>(T value, Func<T, bool> conditionToCheck, Func<T> functio
             return value;
         }
 
+        // Will not print an error message
+        public static async Task PollAsync<T>(T value, Func<T, bool> conditionToCheck, Func<Task<T>> functionToCall, int timeout, ILogger logger)
+        {
+            ValidateTimeoutValue(timeout, logger);
+            DateTime startTime = DateTime.Now;
+
+            while (conditionToCheck(value))
+            {
+                if (functionToCall != null)
+                {
+                    value = await functionToCall();
+                }
+
+                CheckIfTimedOut(startTime, timeout, logger);
+                await Task.Delay(1000);
+            }
+        }
+
+        // Will not print an error message
+        public static async Task PollAsync<T>(T value, Func<T, bool> conditionToCheck, Func<T, Task<T>> functionToCall, int timeout, ILogger logger)
```

Comment:
Duplicate ?

Reply:
No, this is an overload. One is Func<Task<T>>, this one has Func<T, Taks<T>>